src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,491 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod conversion;
+
+use std::fmt::{self, Display, Formatter};
+
+use crate::connectors::prelude::*;
+
+use clickhouse_rs::{
+    errors::Error as CError,
+    types::{DateTimeType, SqlType},
+    Block, ClientHandle, Pool,
+};
+use simd_json::StaticNode;
+
+#[derive(Default, Debug)]
+pub(crate) struct Builder {}
+
+#[async_trait::async_trait]
+impl ConnectorBuilder for Builder {
+    fn connector_type(&self) -> ConnectorType {
+        "clickhouse".into()
+    }
+
+    async fn build(&self, alias: &str, raw_config: &ConnectorConfig) -> Result<Box<dyn Connector>> {
```

Comment:
we can use `build_cfg` here (it's new) and save the error case for a missing config

Reply:
Thanks for adding this function! Commit [8b41100](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/8b41100aab463fc4bf8aec3c36572579eedc1d36) should address this.